src/DataTable/README.md

```diff
@@ -707,4 +707,220 @@ For a more desktop friendly view, you can move filters into a sidebar by providi
     <DataTable.EmptyTable content="No results found" />
     <DataTable.TableFooter />
   </DataTable>
-```
\ No newline at end of file
+```
+
+### Expandable rows
+`DataTable` supports expandable rows which once expanded render additional content under the row.
+`DataTable` will automatically render a separate column with expand toggle for each row that has a `subcomponent` defined, optionally it will render a header cell for the column with expand / collapse all rows behaviour, see examples below.
+
+#### Default view
+```jsx live
+  <DataTable 
+    showExpandAll
+    itemCount={7}
+    data={[
+      {
+        name: 'Lil Bub',
+        color: 'brown tabby',
+        famous_for: 'weird tongue', 
+        subcomponent: "This is just a string"
```

Comment:
The `data` prop generally represents the set of data to display in the table, and often can be thought of a list of results from an API response. That said, it doesn't quite make sense to incorporate `subcomponent` into the `data` prop since `data` is generally dynamic.

Rather, I think we should try to incorporate more of the pattern laid out in the `react-table` documentation for expanding subcomponents, where all rows would use the same subcomponent, but with row data passed in by `react-table`. See this[ documentation example](https://react-table.tanstack.com/docs/examples/sub-components).

e.g.,

```jsx
/**
 * example custom sub component for use on all expandable rows.
 * note that it has access to the `row` data so it could display
 * additional data from the row or make any additional API call(s) using the row data.
 */
const renderRowSubComponent = ({ row }) => (
  <pre>
    <code>{JSON.stringify({ values: row.values }, null, 2)}</code>
  </pre>
);
```

```jsx
/**
 * example subcomponent exported from `DataTable` for use by consumers.
 */
export const ExpandRow = ({ row }) => {
  return (
    <span {...row.getToggleRowExpandedProps()}>
      {row.isExpanded ? '-' : '+'}
    </span>
  );
};

DataTable.ExpandRow = ExpandRow;
```

```jsx
/**
 * example usage of DataTable with expanding subcomponents
 */
<DataTable
  isExpandable
  renderRowSubComponent={renderRowSubComponent}
  columns={[
    {
      id: 'expander',
      Header: () => null, // or optionally a custom/exported "Expand all" subcomponent
      Cell: DataTable.ExpandRow,
    },
    {
      id: 'name',
      Header: 'Name',
      accessor: 'name',
    },
  ]}
  data={[
    {
      name: 'Lil Bub',
      color: 'brown tabby',
      famous_for: 'weird tongue', 
    },
    {
      name: 'Grumpy Cat',
      color: 'siamese',
      famous_for: 'serving moods',
    },
  ]}
/>
```

Then, in `DataTable's` `TableRow` component, an additional `tr` could be rendered when `row.isExpanded` is true, calling `renderRowSubComponent` which passed the `row` argument.